### PR TITLE
fix call to std::isfinite() on Windows

### PR DESCRIPTION
**Pull request by apwojcik**
The standard library from MSVC does implement `std::infinite()` for floating-point types only. The function is used internally by many other math functions, such as `std::isnan()`. The C++ specification mentions optional overloads for integer types. However, there are no additional overloads for integer types in the standard library from MSFT, and according to the C++ specification, all integer types should be treated as doubles. So, in the presence of overloaded functions, the overload function casts the integer type to double and calls the floating-point version of a function. Reference: https://en.cppreference.com/w/cpp/numeric/math/isfinite

Long story short: without the `static_cast<>`, the MIGraphX library is not compiling on Windows.

### CODE REVIEW

**Review by umangyadav — APPROVED** [review]

**Review by pfultz2 — APPROVED** [review]

**Review by TedThemistokleous — APPROVED** [review]

File: src/program.cpp [hunk 1]
```diff
@@ -347,7 +347,7 @@ void program::finalize()
 template <class T>
 std::string classify(T x)
 {
-    switch(std::fpclassify(x))
+    switch(std::fpclassify(static_cast<double>(x)))
```
Comment (lakhinderwalia) [inline src/program.cpp:350]: This breaks for a subnormal float conversion to double. It won't stay subnormal anymore. Thanks.
